src/custodia/cli/__init__.py

```diff
@@ -58,18 +58,32 @@ def server_check(arg):
     return 'http+unix://{}'.format(url_escape(arg, ''))
 
 
+def instance_check(arg):
+    if set(arg).intersection(':/@'):
+        raise argparse.ArgumentTypeError(
+            'Instance name contains invalid characters')
+    return arg
+
+
 def split_header(arg):
     name, value = arg.split('=')
     return name, value
 
 
-main_parser.add_argument(
+group = main_parser.add_mutually_exclusive_group()
+group.add_argument(
     '--server',
     type=server_check,
-    default='/var/run/custodia/custodia.sock',
```

Comment:
I think we should keep support for the old defaults as an "implicit" instance, or we'll break upgrades.